### PR TITLE
Add arguments package generated by SWIG tool

### PR DESCRIPTION
**Pull request by v-thakkar**
In preparation for embedding async profiler, we need wrapper GO API around C++ code that helps us interact with the async profiler functionality. This API is meant to be used by a parca-agent package that will provide similar functionality as [profiler.sh](https://github.com/async-profiler/async-profiler/blob/master/profiler.sh) to do a bunch of actions like `start`, `stop`, `resume` with custom configured options: `event`, `cstack`, `interval`, `format` . This go package will be linked with `libasyncprofiler.so`.

This PR adds the wrapper code generated by the SWIG tool. Please see README and comments in the interface file for more details about the package.

Notes: 
- SWIG-generated code doesn't follow a bunch of idiomatic Go standards so some linters are disabled for this package.
- arguments header file is added for the reference.

### CODE REVIEW

**Comment by v-thakkar** [comment]
This PR is ready to be reviewed. Note that the users of the package will be added in the followup PR. It is planned to be used by the new profiler package - asyncjava in[ this](https://github.com/v-thakkar/parca-agent/commit/d50119933f92fa91bf85c06020168ef078603952#diff-22d63ea516a1fbf04cf659d7f28054a1260fd2caad8f77e622426469c2a1b429) WIP branch.

**Comment by brancz** [comment]
Very cool! lgtm

I assume we'll only be able to add a real unit test once we actually also add the async profiler shared object right? If so I'm happy to wait for a test in that PR.

File: .golangci.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ linters:
     - varnamelen
     - wrapcheck
     - wsl
+    - interfacebloat
```
Comment (brancz) [inline .golangci.yml:42]: is it possible to just ignore this one for the `pkg/arguments` package?

**Review by brancz — APPROVED** [review]
Awesome!

**Pull request merged**